src/scripts/actions/VideoActions.js

```diff
@@ -3,22 +3,38 @@
 import { createAction } from 'redux-actions'
 
 import paratii from 'utils/ParatiiLib'
-import { VIDEO_SELECT, INIT_VIDEOSTORE } from 'constants/ActionConstants'
+import {
+  VIDEO_SELECT,
+  INIT_VIDEOSTORE,
+  VIDEOFECTH_ERROR,
+  VIDEOFECTH_SUCCESS
+} from 'constants/ActionConstants'
 import VideoRecord from 'records/VideoRecords'
 
 import type { Dispatch } from 'redux'
 
 export const selectVideoAction = createAction(VIDEO_SELECT)
 export const initVideoStore = createAction(INIT_VIDEOSTORE)
+export const videoFetchError = createAction(VIDEOFECTH_ERROR)
+export const videoFetchSuccess = createAction(VIDEOFECTH_SUCCESS)
 
 export const fetchVideo = (id: string) => async (dispatch: Dispatch<*>) => {
-  const videoInfo = await paratii.eth.vids.get(id)
-  // TODO: previous line should be replaced with next line once the db is updated
-  // let videoInfo = await paratii.core.vids.get(id)
-
+  let videoInfo
   // update the global state with the fetched information
-  dispatch(selectVideoAction(videoInfo))
-  dispatch(initVideoStore(new VideoRecord(videoInfo)))
+  try {
+    videoInfo = await paratii.eth.vids.get(id)
+
+    // TODO: previous line should be replaced with next line once the db is updated
+    // let videoInfo = await paratii.core.vids.get(id)
+    dispatch(selectVideoAction(videoInfo))
+    dispatch(initVideoStore(new VideoRecord(videoInfo)))
+    dispatch(videoFetchSuccess(videoInfo))
```

Comment:
I think this gives us an opportunity to refine our videos reducer and actions a little bit. 

Instead of having a separate action dispatched for the video data and for the video fetch status, we can update the fetch status directly in the reducer for `initVideoStore`. This should be more efficient since it leads to running through the redux reducer lifecycle one fewer time.

I would also consider renaming `initVideoStore` to `videoFetchSuccess`. 

So, in sum, `initVideoStore` and `videoFetchSuccess` can be combined into one action. If `initVideoStore` is used in other places that adds a bit more complexity but in general I think that is the direction we should head. Thoughts?

Reply:
yes i totally agree with you. 
do we merge this PR and we open a new enhancement issue about this, or i'll modify it here?